### PR TITLE
Split translation part from chainer_compiler.compile

### PR DESCRIPTION
**Pull request by ir5**
This PR changes the interface of `CompiledModel` so that it is split into translation part and compilation part.

I also changed an MNIST example to run export part and compile part separately.

### CODE REVIEW

File: scripts/run-travis-tests.sh [hunk 1]
```diff
@@ -40,8 +40,10 @@ PYTHONPATH=. run runtests ./scripts/runtests.py 2>&1
 PYTHONPATH=. run pytest pytest -sv tests
 PYTHONPATH=. run canonicalizer_tests pytest testcases/elichika_tests/canonicalizer
 
-PYTHONPATH=. run train_mnist python3 examples/mnist/train_mnist.py \
-     -d native --compile -I 3 --use-fake-data
+PYTHONPATH=. run train_mnist_export python3 examples/mnist/train_mnist.py \
+     -d -1 --export /tmp/tmp_mnist_model.onnx -I 3 --use-fake-data
```
Comment (shinh) [inline scripts/run-travis-tests.sh]: Is `-d -1` intentional? s/-1/native/?

File: chainer_compiler/chainer_compiler.py [hunk 2]
```diff
@@ -182,65 +183,51 @@ def backward(self, indexes, flat_gys):
...
 
 
-def _run_translator(translator, mc, inputs):
+def export(model, inputs, filename=None, translator='onnx_chainer'):
     if translator == 'ch2o':
         from chainer_compiler import ch2o
-        xmodel = ch2o.compile_model(mc, inputs)
-        f = tempfile.NamedTemporaryFile(delete=False)
+        xmodel = ch2o.compile_model(model, inputs)
+        if filename is None:
```
Comment (shinh) [inline chainer_compiler/chainer_compiler.py]: Could you factor out line 190-193? Just moving 190-193 and 199-202 to the beginning of this function would be enough

**Comment by shinh** [comment]
ChainerCI, test this please

**Comment by shinh** [comment]
FYI, as per tanaka-san, it's OK to ignore these failures

File: scripts/run-travis-tests.sh (hunk 1 above)
Comment (ir5) [reply scripts/run-travis-tests.sh]: I changed the translator for MNIST example to onnx-chainer, and `native` did not work for onnx-chainer.

File: chainer_compiler/chainer_compiler.py (hunk 2 above)
Comment (ir5) [reply chainer_compiler/chainer_compiler.py]: Sure, I'll do in future PR.

**Pull request merged**